datafusion/core/tests/sqllogictests/src/main.rs

```diff
@@ -138,7 +140,14 @@ fn format_batches(batches: &[RecordBatch]) -> Result<String> {
 }
 
 async fn run_query(ctx: &SessionContext, sql: impl Into<String>) -> Result<String> {
-    let df = ctx.sql(&sql.into()).await.unwrap();
+    let sql = sql.into();
+    // Check if the sql is `insert`
+    if sql.trim_start().to_lowercase().starts_with("insert") {
+        // Process the insert statement
+        insert(ctx, sql).await?;
+        return Ok("".to_string());
+    }
```

Comment:
👍  I like this basic approach (special case the sql and route it to the test runner implementation).

One thing that might be worth doing is to actually try and parse the input into sql, to detect INSERT statements though I think string manipulation is fine too or we could do this later

```rust
    // Handle any test only special case statements
    let sql = sql.into();
    match DFParser::parse_sql(&sql) {
        Ok(Statement(Insert)) => {
        //debug!("Parsed statement: {:#?}", stmt);

        }
        Err(_) => {
            // ignore anything else, including errors -- they will be handled by the sql context below
        }
    };
```

Reply:
Yes, more generic way, will fix in next PR.